### PR TITLE
Feature/signup

### PR DESCRIPTION
**Pull request by HuongSakada**
In this feature/signup is to create a new user and then transfer 100riel and $100 from admin account to this new account.
The user needs to input Firstname, Lastname, and Username with validation. After, the user input the valid data, the system create key-pairs for creating a new account and return the private key to the user to download. The system also alerts the user about the success or failure of this process.

### CODE REVIEW

File: docker/iroha/genesis.block [hunk 1]
```diff
@@ -83,6 +79,20 @@
                               "accountId": "admin@iroha",
                               "roleName": "admin"
                            }
+                        },
+                        {
+                           "createAsset": {
+                              "assetName": "riel",
+                              "domainId": "iroha",
+                              "precision": 10
```
Comment (Asmadek) [inline docker/iroha/genesis.block]: Are we really need such precision? Same for usd#iroha

File: src/components/Login.vue [hunk 2]
```diff
@@ -51,14 +51,16 @@
           type="danger"
           @click="onSubmit"
           :loading="isLoading">
-          LOGIN
+          LOG IN
         </el-button>
       </el-form-item>
 
+      <p class="auth-container-info">Don’t have an account?</p>
```
Comment (Asmadek) [inline src/components/Login.vue]: What is the reason to use <p> here? If there is no specific reason, better to use <div>

File: src/components/Signup.vue [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      <el-form-item class="login-button-container">
+        <el-button
+          class="fullwidth"
+          @click="onSubmit"
+          type="danger">
+          Register
+        </el-button>
+      </el-form-item>
+
+      <p class="auth-container-info">Already have an account</p>
```
Comment (Asmadek) [inline src/components/Signup.vue]: Same as previous

File: src/components/Signup.vue [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    </el-form>
+  </div>
+
+  <el-dialog
+    title="Your private key"
+    :close-on-click-modal="false"
+    :visible.sync="dialogVisible"
+    width="450px"
+    center>
+    <div >Download your private key and keep it secret!</div>
```
Comment (Asmadek) [inline src/components/Signup.vue]: Unnecessary space

File: src/store/Account.js [hunk 5]
```diff
@@ -248,6 +257,61 @@ const actions = {
       .catch(err => {
         commit(types.TRANSFER_ASSET_FAILURE, err)
       })
+  },
+
+  createAccount ({ commit }, { accountName, domainId }) {
+
```
Comment (Asmadek) [inline src/store/Account.js]: Unnecessary  empty line

File: src/store/Account.js [hunk 6]
```diff
@@ -248,6 +257,61 @@ const actions = {
...
+      )
+      .catch(err => {
+        throw err
+      })
+    });
+
+    return Promise.all(setAccountDetailArray)
+  },
+
+  initialAssetQuantityForUser ({ state }, { accountId }) {
```
Comment (Asmadek) [inline src/store/Account.js]: It is function, but called like variable. Functions should be called like action with verb.

File: src/utils/functions.js [hunk 7]
```diff
@@ -1,9 +1,46 @@
-import { cache } from './util'
+import { cache, newPreCommandServiceOptions } from './util'
+import commands from 'iroha-helpers/lib/commands'
 
 function isLoggedIn () {
   return !!cache.username
 }
 
+function preDefinedPrivateKey () {
```
Comment (Asmadek) [inline src/utils/functions.js]: It's function, but return data. It's okay to use such name for such functions.

File: src/utils/functions.js [hunk 8]
```diff
@@ -1,9 +1,46 @@
...
+  return ['0f0ce16d2afbb8eca23c7d8c2724f0c257a800ee2bbd54688cec6b898e3f7e33']
+}
+
+function preDefinedAccountId () {
+  return 'admin@iroha'
+}
+
+async function preInitialAssetQuantityForUser ({ assetId, accountId }) {
```
Comment (Asmadek) [inline src/utils/functions.js]: Same situation like with `initialAssetQuantityForUser`

File: src/utils/functions.js (hunk 7 above)
Comment (Asmadek) [reply src/utils/functions.js]: But i think better to use predefined (as one word) not preDefined (as two words)

File: src/utils/util.js [hunk 9]
```diff
@@ -25,6 +27,16 @@ export function newCommandServiceOptions (privateKeys, quorum) {
   }
 }
 
+export function newPreCommandServiceOptions () {
```
Comment (Asmadek) [inline src/utils/util.js]: newPredefinedCommandServiceOptions looks better 
or
newCommandServiceOptionsWithPredefinedUser

**Review by Asmadek — CHANGES_REQUESTED** [review]

File: docker/iroha/genesis.block (hunk 1 above)
Comment (HuongSakada) [reply docker/iroha/genesis.block]: I think we do not need such the same precision between riel#iroha and usd#iroha but I do not know what is the suitable precision for riel#iroha. It should be 2 right?

File: src/components/Login.vue (hunk 2 above)
Comment (HuongSakada) [reply src/components/Login.vue]: To give information to the user about the signup or to give a description of the signup button to the user

File: docker/iroha/genesis.block (hunk 1 above)
Comment (Asmadek) [reply docker/iroha/genesis.block]: According to https://en.wikipedia.org/wiki/ISO_4217 we should use 2 digits.

File: src/components/Login.vue (hunk 2 above)
Comment (Asmadek) [reply src/components/Login.vue]: Oh, I forgot that tag names doesn't shown in comments. 

What is the reason to use `<p>` here? If there is no specific reason, better to use `<div>`

**Comment by Asmadek** [comment]
Approve

**Review by Asmadek — APPROVED** [review]

**Pull request merged**